lib/useButter.js

```diff
@@ -0,0 +1,225 @@
+     * 
+     * @async
+     * @function get
+     * @param {string} [apiEndpoint=BASE_PATH] - The API endpoint to send the request to.
+     * @param {Object} [localParams={}] - Additional parameters to include in the request.
+     * @returns {Promise<Object>} An object containing the data, meta, and any errors from the API response.
+     * @throws Will throw an error object containing errors, config, and params if the API response includes errors.
+     */
+    async function get (apiEndpoint = BASE_PATH, localParams = {}) {
+        // Extract apiToken and separate it from the rest of the user's configuration
+        const {
+            apiToken: auth_token,
+            ...userConfig
+        } = butterConfig;
+        
+        // Initialize headers for the request
+        const butterHeaders = new Headers({
+            "Content-Type": "application/json",
+            "X-Butter-Client": `JS/${ BUTTER_CLIENT_VERSION }`
+        });
+        
+        // Append additional header for server-side environments
+        if (typeof window === "undefined") {
+            butterHeaders.append("Accept-Encoding", "gzip");
+        }
+        
+        // Prepare the request configuration using the onRequest hook
+        const {
+            config,
+            headers,
+            params
+        } = await useOnRequest(
+            apiEndpoint,
+            auth_token,
+            {
+                cancelRequest,
+                config: userConfig,
+                headers: butterHeaders,
+                params: localParams,
+                type,
+            }
+        );
+        
+        try {
+            // Set up fetch timeout and signal
+            const {
+                cleanup,
+                signal
+            } = useFetchTimeout(fetchController, friendlyType, config.timeout);
+
+            // Perform the API request
+            const response = await fetch(
+                `${ apiEndpoint }?${ new URLSearchParams(params) }`,
+                {
+                    cache: config.cache,
+                    method: "GET",
+                    headers,
+                    signal
+                }
+            );
+            
+            // Clean up after the fetch timeout
+            cleanup();
+            
+            // Process the response using the onResponse hook
+            const {
+                data,
+                meta,
+                ...errors
+            } = await useOnResponse(
+                response,
+                {
+                    config,
+                    // pass the request as user won't see our
+                    // changes for auth_token or testMode
+                    // remove testMode params and auth_token in onResponse
+                    params,
+                    type
+                }
+            );
+            
+            // Check for errors in the response and throw if present
+            if (Object.keys(errors).length) {
```

Comment:
Are we 100% sure that `errors` is not undefined?

Reply:
Yes. The `errors` object is returned and defined by the Butter API. Any browser/fetch/404/500 errors are caught before reaching this point, so we are only looking for actual errors as sent by the server in the data repsonse.